consensus/core/src/tx.rs

```diff
@@ -30,27 +35,57 @@ pub use smallvec::smallvec as scriptvec;
 pub type ScriptPublicKeys = HashSet<ScriptPublicKey>;
 
 /// Represents a Kaspad ScriptPublicKey
-#[derive(Default, Debug, PartialEq, Eq, Serialize, Clone, Hash)]
-#[serde(rename_all = "camelCase")]
+#[derive(Default, Debug, PartialEq, Eq, Clone, Hash)]
 pub struct ScriptPublicKey {
     version: ScriptPublicKeyVersion,
     script: ScriptVec, // Kept private to preserve read-only semantics
 }
 
+#[derive(Default, Debug, PartialEq, Eq, Serialize, Deserialize, Clone, Hash)]
+#[serde(rename_all = "camelCase")]
+struct ScriptPublicKeyInternal<'a> {
```

Comment:
```suggestion
#[derive(Default, Debug, PartialEq, Eq, Serialize, Deserialize, Clone, Hash)]
#[serde(rename_all = "camelCase")]
#[serde(rename = "ScriptPublicKey")]
struct ScriptPublicKeyInternal<'a> {
```

Could you add this renaming? It was my fault. I think without renaming it xml and other structure-name sensitive formats can work incorrectly

Reply:
fixed